src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsMigrationService.java

```diff
@@ -0,0 +1,70 @@
+package de.tum.in.www1.artemis.service.connectors.jenkins;
+
+import static de.tum.in.www1.artemis.config.Constants.NEW_RESULT_RESOURCE_API_PATH;
+
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.xml.transform.TransformerException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+import org.w3c.dom.Document;
+
+import de.tum.in.www1.artemis.domain.VcsRepositoryUrl;
+import de.tum.in.www1.artemis.exception.JenkinsException;
+import de.tum.in.www1.artemis.service.connectors.ci.CIMigrationService;
+import de.tum.in.www1.artemis.service.connectors.jenkins.jobs.JenkinsJobService;
+import de.tum.in.www1.artemis.service.util.XmlFileUtils;
+
+/**
+ * Services for executing migration tasks for Jenkins.
+ */
+@Service
+@Profile("jenkins")
+public class JenkinsMigrationService implements CIMigrationService {
+
+    private final Logger log = LoggerFactory.getLogger(JenkinsMigrationService.class);
+
+    @Value("${server.url}")
+    private String artemisServerUrl;
+
+    private final JenkinsJobService jenkinsJobService;
+
+    public JenkinsMigrationService(JenkinsJobService jenkinsJobService) {
+        this.jenkinsJobService = jenkinsJobService;
+    }
+
+    @Override
+    public void overrideBuildPlanNotification(String projectKey, String buildPlanKey, VcsRepositoryUrl repositoryUrl) {
```

Comment:
Maybe I missed it: Is there a test that checks that the replacement regex is working as intended with a few examples of to-be-expected XML contents?

Reply:
Not necessarily. We would need to expose private methods just for testing purposes. I'm not sure how we could work around that.

Reply:
I meant it as: mocking `getJobConfig` to return a known XML, let the private method perform the replacement, checking the new XML argument passed to `jenkinsJobService.updateJob`. That would not require access to `private`s.

Reply:
You mind if I do this in a follow-up?